lib/galaxy/model/__init__.py

```diff
@@ -8376,7 +8376,9 @@ def to_dict(self, view="collection", value_mapper=None, step_details=False, lega
             for output_assoc in self.output_dataset_collections:
                 label = output_assoc.workflow_output.label
                 if not label:
-                    continue
+                    label = (
+                        label
+                    ) = f"{output_assoc.workflow_output.output_name} (Step {output_assoc.workflow_output.workflow_step.order_index + 1})"
```

Comment:
@mvdbeek I suppose this can just be:

```python
                    label = f"{output_assoc.workflow_output.output_name} (Step {output_assoc.workflow_output.workflow_step.order_index + 1})"
```

?

Reply:
yes, I don't know what happened here 🤷